### DISCUSSION TITLE
Dynamic images, looking for help.

### DISCUSSION THREAD

**Discussion by jamiecarter7** [nuxt/nuxt#19617] [Questions] [answered]
Thanks for reading!

In a nut shell:
- I'm attempting to upgrade from Vue2 to Nuxt 3
- I have several components that use require(*dynamic-image-url*)
- I am hitting a wall trying to work out how to change my code for this to work.

My problem:
- I have a component that is using JSON to generate 400+ cards, it uses the data to select the relevant paths for each image file. Each component only uses around 8 of the 400 images. But the combinations are done programmatically from the JSON.

What I've tried to so far:
- Lots of reading on the meta .glob stuff with vite (it seems this is not a good path to go)
- I migrated to using /public folder, but now on deploying the baseURL option is not applying to my public paths so my website is at '/spirespy' and my urls of all my images point to '/' so they are broken.
- Tried to look into image hosting but not sure where to find good documentation.

I am feeling at a dead end with this - been researching all kinds of things to work out a way around this but in the last few days I feel stuck.

Questions:
What is the vite way to tackle this issue?
Should I be importing all 400+ images into my component and then trying to select the right one? Would this trigger a load on all 400+ images whenever the component uses just 8?
I have my dev env working perfectly using the original code and /public folder - but there is an issue when I hope at a different subdomain '/spirespy' - baseURL is not being appended to those urls. Is this something I need to do manually with all my image links inside the component?

I've been attempting to google 100 solutions to this over the last 2 weeks - this feels like a last resort ask for help.

Thanks again

**Comment by TheAlexLichter** [comment, marked as answer]
See https://github.com/nuxt/nuxt/issues/14766#issuecomment-1397365205 for a thorough solution ☺️

**Reply by jamiecarter7** [reply]
Thank you! I am so happy to get this fixed
